src/main/java/hudson/plugins/logparser/LogParserPublisher.java

```diff
@@ -19,18 +21,21 @@
 import java.util.logging.Level;
 import java.util.logging.Logger;
 
+import jenkins.tasks.SimpleBuildStep;
 import net.sf.json.JSONObject;
 
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
 import org.kohsuke.stapler.StaplerRequest;
 
-public class LogParserPublisher extends Recorder implements Serializable {
+public class LogParserPublisher extends Recorder implements SimpleBuildStep, Serializable {
     private static final long serialVersionUID = 1L;
-    public final boolean unstableOnWarning;
-    public final boolean failBuildOnError;
-    public final boolean showGraphs;
-    public final String parsingRulesPath;
-    public final boolean useProjectRule;
-    public final String projectRulePath;
+    public boolean unstableOnWarning;
+    public boolean failBuildOnError;
+    public boolean showGraphs;
+    public String parsingRulesPath;

```

Comment:
I don't know what you mean @oleg-nenashev. This is the common approach in many plugins since `@DataBoundSetter` was added to the API. Are you complaining about attributes being `public` or something?